packages/core/core/src/Parcel.js

```diff
@@ -381,12 +382,14 @@ export default class Parcel {
             type: e.type,
             path: toProjectPath(resolvedOptions.projectRoot, e.path),
           })),
+          {isInitialBuild: this.#isInitialBuild},
```

Comment:
I don't think you'll need this flag at all. It should only run in the below case handled by loadRequestGraph